### ISSUE TITLE
Bug related to transp

### ISSUE THREAD

**Issue by nad** [agda/agda#5579]
The following code is accepted by Agda 2.6.2, but rejected by a recent development version:
```agda
{-# OPTIONS --cubical #-}

open import Agda.Builtin.Cubical.Path
open import Agda.Primitive.Cubical

postulate
  Index : Set
  i     : Index

data D : Index → Set where
  c : D i

cong : {A B : Set} (x y : A) (f : A → B) → x ≡ y → f x ≡ f y
cong _ _ f x≡y = λ i → f (x≡y i)

refl : {A : Set} {x : A} → x ≡ x
refl {x = x} = λ _ → x

subst :
  {A : Set} {x y : A}
  (P : A → Set) → x ≡ y → P x → P y
subst P eq p = primTransp (λ i → P (eq i)) i0 p

postulate
  subst-refl :
    {A : Set} {x : A} (P : A → Set) {p : P x} →
    subst P refl p ≡ p

f : {i : Index} (xs : D i) → D i
f c = c

works : f (subst D refl c) ≡ c
works = cong (subst D refl c) _ f (subst-refl D)

fails : f (subst D refl c) ≡ c
fails = cong _ _ f (subst-refl D)
```
```
f (Bug.transpD (λ i₁ → refl i₁) i0 _y_74) !=
Bug.transpX-D
(λ n →
   primComp (λ i₁ → Index)
   (λ i₁ .o →
      primPOr i0
      (primIMax (primINeg (primINeg n)) (primIMax (primINeg n) i0))
      (λ .o₁ → i)
      (primPOr (primINeg (primINeg n)) (primIMax (primINeg n) i0)
       (λ .o₁ →
          primComp (λ i₂ → Index)
          (λ i₂ .o₂ →
             primPOr i0
             (primIMax (primINeg (primINeg i₁)) (primIMax (primINeg i₁) i0))
             (λ .o₃ → i)
             (primPOr (primINeg (primINeg i₁)) (primIMax (primINeg i₁) i0)
              (λ .o₃ → refl i₂) (λ .o₃ → i))
             _)
          i)
       (λ .o₁ → i))
      _)
   i)
i0 c
of type D i
when checking that the expression cong _ _ f (subst-refl D) has
type f (subst D refl c) ≡ c
```
Note the difference between `works` and `fails`:
```agda
works = cong (subst D refl c) _ f (subst-refl D)
fails = cong _ _ f (subst-refl D)
```
Is the problem that some part of the type-checking process is not postponed when a meta-variable is encountered?

**Comment by nad** [comment]
@Saizan, could you write something about what the problem was?

**Comment by Saizan** [comment]
It was about the `useInjectivity` machinery triggering for `f`:  the term `f (Bug.transpD (λ i₁ → refl i₁) i0 _y_74)` is blocked on the meta `_y_74` so in general we should postpone its comparison to other terms.
Agda however recognised `f` as invertible so it kept going and then failed from there.
